modules/check_in/spec/services/v2/chip/service_spec.rb

```diff
@@ -176,6 +176,50 @@
     end
   end
 
+  describe '#pre_check_in' do
+    let(:uuid) { 'd602d9eb-9a31-484f-9637-13ab0b507e0d' }
+    let(:params) do
+      {
+        demographics_up_to_date: true,
+        next_of_kin_up_to_date: true,
+        check_in_type: :pre_check_in
+      }
+    end
+
+    context 'when token is already present' do
+      let(:resp) { 'Pre-checkin successful' }
+
+      before do
+        allow_any_instance_of(::V2::Chip::Service).to receive(:token).and_return('jwt-token-123-abc')
+        allow_any_instance_of(::V2::Chip::Client).to receive(:pre_check_in)
+          .and_return(Faraday::Response.new(body: 'Pre-checkin successful', status: 200))
+      end
+
+      it 'returns correct response' do
+        expect(subject.build(check_in: valid_check_in, params: params)
+                      .pre_check_in.body).to eq(resp)
+      end
+    end
+
+    context 'when token is not present' do
+      let(:resp) { { permissions: 'read.none', status: 'success', uuid: uuid } }
+
+      before do
+        allow_any_instance_of(::V2::Chip::Service).to receive(:token).and_return(nil)
+      end
+
+      it 'returns unauthorized message in response body' do
+        expect(subject.build(check_in: valid_check_in, params: params)
+                      .pre_check_in.body).to eq(resp.to_json)
+      end
+
+      it 'returns unauthorized status' do
+        expect(subject.build(check_in: valid_check_in, params: params)
+                      .pre_check_in.status).to eq(401)
+      end
+    end
+  end
+
   describe '#token' do
```

Comment:
Can we please add tests for ``demographic_confirmations`` method?

Reply:
As part of rpsec, I feel that this method is being called within service spec and there's not much else to test. It probably makes more sense to test it in some kind of integration test (which we don't have currently) to make sure that what we get from FE through the controller POST params is translated correctly to the CHIP service. 

Do you still feel rspec for this are needed, and if so what would you suggest testing in this method?

Reply:
Since we have a ternary condition check, thought is good to have tests like,
- `demographics_needs_update` set with true when FE sends `check_in_body[:demographics_up_to_date]` with false 
- `demographics_needs_update` set with false when FE sends `check_in_body[:demographics_up_to_date]` with true 
- `next_of_kin_needs_update` set with true when FE sends `check_in_body[:next_of_kin_up_to_date]` with false
- `next_of_kin_needs_update` set with false when FE sends `check_in_body[:next_of_kin_up_to_date]` with true

Reply:
Are we testing here that a ternary condition works?

Reply:
Yes 👍

Reply:
So in future this rspec will make sure that this contract between FE/vets-api/CHIP doesn't change in check-in module

Reply:
I honestly feel that it's superfluous and unnecessary to test that ternary conditions in Ruby works. Like I said earlier, it makes more sense for an integration test to validate the input and output and since we don't have an automated integration test approach currently, it will be a part of integration testing we'll do as soon as FE code is ready.

Reply:
Synced offline, above tests will be covered in integration test.